nucypher/blockchain/eth/interfaces.py

```diff
@@ -825,13 +826,18 @@ def deploy_contract(self,
                       f"deployer address {deployer_address} "
                       f"and parameters {pprint_args}")
 
-        transaction_function = contract_factory.constructor(*constructor_args, **constructor_kwargs)
+        constructor_function = contract_factory.constructor(*constructor_args, **constructor_kwargs)
+        constructor_calldata = encode_constructor_arguments(self.client.w3,
+                                                            constructor_function,
+                                                            *constructor_args,
+                                                            **constructor_kwargs)
+        self.log.info(f"Constructor calldata: {constructor_calldata}")
```

Comment:
Should this string just replace the `pprint_args` one used on L827 (i.e. move that one here and replace `pprint_args` with `constructor_calldata`, or do we need both log entries?

Reply:
We need both entries because they're quite different:
```
2020-10-01T19:00:48+0200 [test-blockchain#info] Deploying contract StakingInterface:v1.6.1 with deployer address 0x7E5F4552091A69125d5DfCb7b8C2659029395Bdf and parameters 0x2946259E0334f33A064106302415aD3391BeD384, 0x51a240271AB8AB9f9a21C82d9a85396b704E164d, 0xA3183498b579bd228aa2B62101C40CC1da978F24, 0x7758F98C1c487E5653795470eEab6C4698bE541b
2020-10-01T19:00:48+0200 [test-blockchain#info] Constructor calldata: 0x0000000000000000000000002946259e0334f33a064106302415ad3391bed38400000000000000000000000051a240271ab8ab9f9a21c82d9a85396b704e164d000000000000000000000000a3183498b579bd228aa2b62101c40cc1da978f240000000000000000000000007758f98c1c487e5653795470eeab6c4698be541b
```